File: HomeWork1/Hw1.Tests/ProgramTests.cs
using Xunit;

namespace Hw1.Tests;

public class ProgramTests
{
    [Theory]
    [InlineData("1 + 2")]
    [InlineData("1 / 2")]
    [InlineData("1 - 2")]
    [InlineData("0 / 0")]
    [InlineData("1 / 0")]
    public void ProgramDoesNotFallOnMathOperations(string input)
    {
        RunTest(input, 0);
    }

    [Theory]
    [InlineData("1 2")]
    [InlineData("1")]
    [InlineData("1 . 3 5")]
    [InlineData("1 +")]
    public void ProgramCatchesWrongNumberOfArguments(string input)
    {
        RunTest(input, -2);
    }

    [Theory]
    [InlineData(". + 2")]
    [InlineData("a - f")]
    public void ProgramCatchesNonDoubleInput(string input)
    {
        RunTest(input, -2);
    }

    [Theory]
    [InlineData("3 . 4")]
    public void ProgramCatchesWrongOperations(string input)
    {
        RunTest(input, -3);
    }

    private static void RunTest(string input, int expectedCode)
    {
        var result = Program.Main(input.Split());
        Assert.Equal(expectedCode, result);
    }
}

File: Homework5/Hw5.Tests/ProgramTests.fs
module Hw5.Tests.ProgramTests

open Hw5
open Program
open Xunit        

[<Theory>]
[<InlineData("1", "+","1")>]
[<InlineData("1","-","1")>]
[<InlineData("1","*", "1")>]
[<InlineData("1.0","/","1")>]
[<InlineData("a","+","1")>]   
[<InlineData("1.0",".","1")>]        
[<InlineData("1","/","0")>]        
let test3args(a1, a2, a3) =
    let args = [|a1;a2;a3|]
    Assert.Equal(0, main args)
    
[<Fact>]
let testNull =
    Assert.Equal(0, main null)